Makefile

```diff
@@ -98,14 +98,15 @@ clean-cache:
 # plugins build targets
 #
 
-$(BIN_PATH)/%: plugins/%/*
-	$(Q)echo "Building $@..."; \
-	cd $(dir $<) && $(GO_BUILD) -o $@ .
+build-plugin-%:
+	$(Q)echo "Building plugin $*..."; \
+	cd plugins/$* && $(GO_BUILD) -o $(BIN_PATH)/$* .
 
-$(BIN_PATH)/wasm: plugins/wasm/
-	$(Q)echo "Building $@..."; \
+.PHONY: build-plugin-wasm
+build-plugin-wasm:
+	$(Q)echo "Building plugin wasm..."; \
 	mkdir -p $(BIN_PATH) && \
-	cd $(dir $<) && GOOS=wasip1 GOARCH=wasm $(GO_BUILD) -o $@ -buildmode=c-shared .
+	cd plugins/wasm && GOOS=wasip1 GOARCH=wasm $(GO_BUILD) -o $(BIN_PATH)/wasm -buildmode=c-shared .
```

Comment:
FWIW, current versions of Go also allows for a `-C <path>` argument, which can safe some headaches on having to `cd` to a directory ("and back again" in some cases).

From `go help build`;

```
The build flags are shared by the build, clean, get, install, list, run,
and test commands:

	-C dir
		Change to dir before running the command.
		Any files named on the command line are interpreted after
		changing directories.
		If used, this flag must be the first one in the command line.
	-a
		force rebuilding of packages that are already up-to-date.
```